### ISSUE TITLE
Recover "MOVEMENT" and "DIRECTION" states that have been erased

### ISSUE THREAD

**Issue by patrickm02L** [cityofaustin/atd-data-tech#8639]
The CRIS23 update and subsequent re-uploading of all the CR3s has wiped out the Movement and Direction states in the VZBD. We'll need to restore the QA/QC-ed records from 2017 to present of changes that Xavier made. 

Comment on #4582 via @xavierapostol

> I was doing some QA/QC today and noticed several crashes that I had already assigned the "MOVEMENT" and the "DIRECTION" but now states "UNKNOWN" "MOVEMENT" and the original "DIRECTION". Is there a way to ensure that the ETL process does not automatically revert the values here?
> 
> Here are some examples:
> https://visionzero.austin.gov/editor/#/crashes/18136761
> https://visionzero.austin.gov/editor/#/crashes/18322187
> https://visionzero.austin.gov/editor/#/crashes/18344837
> https://visionzero.austin.gov/editor/#/crashes/18345071
> https://visionzero.austin.gov/editor/#/crashes/18562169

**Comment by patrickm02L** [comment]
Discussion from 3/3/22 Sprint Review about the [breadth of Movement and Direction changes](https://web.microsoftstream.com/video/6801feef-e34f-4f5d-985a-02947950b2cb?st=2592) that need to be changed back to Xavier's QA/QC. He's identified hundreds going back to January 1, 2017 that were reverted to their original state with CRIS23 updates.

**Comment by patrickm02L** [comment]
@chiaberry and I met with @xavierapostal on 3/9/22. 

We looked at a PowerBI dashboard he created which showed 417 records that may have been reverted back to before they were QA/QC-ed. 

**Filters**
- is not Level 5
- unit_1_dir is not, EAST, NORTH, SOUTH, or WEST
- unit_1_mov is UNKNOWN

He updated a few records on the call which he saved and refreshed to show they saved, but when he went back later the Movement and Direction was not saved nor was is showing in the Change Log. 

Xavier recommended looking back at when the Movement and Direction feature was created through when the records started updating after the CRIS23 update to get a sense of the timeline when the problem started to occur. Direction and Movement feature was created 2/2/20, and new records from CRIS23 began updating 1/9/22.

**Related Issues**
- #1581 VZD: Add "direction of movement" column to Unit table
- #4132 VZE Bug - Direction and Movement fields in Unit Cards
- #8140 [Bug ] ~4,200 unexpected crash records were pushed to VZ by CRIS on 2021-01-11

**Comment by mateoclarke** [comment]
Hey @patrickm02L, I edited your comment above to add more clarification. Hope thats ok.

**Comment by patrickm02L** [comment]
@mateoclarke yep, definitely. Thanks for the clarity!

**Comment by mateoclarke** [comment]
@frankhereford, can you post here the outline of strategy we talked about today in the meeting? And one of the whiteboards?

**Comment by frankhereford** [comment]
@mateoclarke - you bet! Here's a recap from my notes.

* From our previous meetings notes, I have written down the action items of:
  * Spot check the UPDATE statements
  * Remove the CRIS provided `veh_trvl_dir_id` from the fields we would be trying to correct
* Additionally, it was recalled that we had agreed to take a different approach of only comparing the units table from prior to the CRIS reprocessing to the current one, and use the differences of that to provide a set of potential corrections.
  *  To that end, I am working on a script which does that comparison, and produces the set of potential restorations.
    * For every change found, it seeks to find the pair of change-log entries which provide the time of the change (which allows us to make an educated guess if it's human initiated or airflow initiated. We will exclude the CRIS initiated changes.
    * For every change found, it will also check to see if the crash occurred after the onset of the CRIS reprocessing, and exclude that set.

Here's a snap of our whiteboard drawings:
![IMG_1482](https://user-images.githubusercontent.com/3653206/161281200-d73bfc84-dac4-4842-99fe-65597de4e675.jpeg)

**Comment by frankhereford** [comment]
A set of ~ 1200 units have been identified as candidates for restoration of the `movement_id` and `travel_direction` field. The crashes related to these units have had their geometry pulled, and the majority lay along the high injury roadways. This set has been sent to the VZ team, along with a map depicting of the spatial correlation between the HIR and the set of crashes in question, for review.

**Comment by patrickm02L** [comment]
@frankhereford thanks for sending over to the VZ team for review.

**Comment by frankhereford** [comment]
Xavier responded with the following:

> As far as the proposed idea is concerned, can you filter out any MANUAL (qa_status = 3) that has the a last_updated date after 3/1/2022 and anything that does not have a cardinal_direction=FALSE? I believe a lot of these are manually re-updated or are an issue I need to address. If you can do that, I would say that you can go ahead with the proposed fix.

I will execute his instructions. I appreciate the well considered feedback!

**Comment by frankhereford** [comment]
After generating the set of update statements from the revised set of proposed restorations, I started to check a sample of the changes in the output to make sure they made sense. This led me to find uncover a mistake that I had included in my script which generated the change set. All the proposed restorations were non-operations, setting the value to what was already in the present database.

Upon [correcting the bug](https://github.com/cityofaustin/atd-vz-data/pull/1064/commits/f9044b0abcbe53e7319cfbe45c981a092fee18ee) and taking a second look at how I had logically tied the [criteria for inclusion](https://github.com/cityofaustin/atd-vz-data/pull/1064/files#diff-7b43071089f32f2eb0a34e89c13fe90cda8c2825c19fa29d8aa58105a083b09cR21-R35) in the final set, I re-ran the scripts. This time, we had a much smaller set of potential changes: 21 records all changing the travel direction. Investigating the scale of this change set, it appears that the VZ team has revisited and manually resubmitted data for these crashes of interest, almost all of which lay along high-injury roadways.

I went through each of the proposed changes and inspected the state of the database, the CR3's diagram and narrative, and have identified the changes that need to be applied. The following statements have been executed in production:

```
update atd_txdot_units set travel_direction = 3 where unit_id = 3996306;
update atd_txdot_units set travel_direction = 7 where unit_id = 3996305;
update atd_txdot_units set travel_direction = 7 where unit_id = 3996304;

update atd_txdot_units set travel_direction = 5 where unit_id = 2660405;
update atd_txdot_units set travel_direction = 1 where unit_id = 2660404;

update atd_txdot_units set travel_direction = 7 where unit_id = 2602558;
update atd_txdot_units set travel_direction = 1 where unit_id = 2602546;

update atd_txdot_units set travel_direction = 5 where unit_id = 2606693;
update atd_txdot_units set travel_direction = 1 where unit_id = 2606689;
```
